tf2/src/cache.cpp

```diff
@@ -267,22 +267,18 @@ bool TimeCache::insertData(const TransformStorage& new_data, std::string* error_
       break;
     storage_it++;
   }
+
+  // Allow to replace TF with the same timestamp
   if (storage_it != storage_.end() && storage_it->stamp_ == new_data.stamp_)
   {
-    if (error_str)
-    {
-      std::stringstream ss;
-      ss << "TF_REPEATED_DATA ignoring data with redundant timestamp";
-      *error_str = ss.str();
-    }
-    return false;
+    *storage_it = new_data;
   }
   else
   {
     storage_.insert(storage_it, new_data);
+    pruneList();
```

Comment:
`pruneList` prunes based on time, it appears, so maybe we want to leave it where it was? We might want to prune in either case.

Reply:
Scratch that. It's using only the times in the list, not the current time.